src/Compilers/Core/Portable/DiagnosticAnalyzer/SourceOrNonSourceFile.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
```

Comment:
This change overall seems to oscillate between two different terminology pairs:

- source file / additional file 
- source file / non source file

Think this change would be better if it were consistent on that path. Given our existing code I think the first is more preferable than the second. Given that I would consider naming this `SourceOrAdditionalFile`